File: Backend/API/src/main/java/com/example/localhistory/homework/dto/response/HomeworkAssignmentDTO.java
package com.example.localhistory.homework.dto.response;

import com.example.localhistory.user.dto.response.StudentDTO;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

// Response sent for homework assigned to a specific student, including current completion state.
@Data
public class HomeworkAssignmentDTO {
    private Long id;
    private Long homeworkId;
    private String title;
    private String description;
    private LocalDate dueDate;
    private LocalDateTime assignedAt;
    private LocalDateTime completedAt;
    private Long landmarkId;
    private String landmarkTitle;
    private Boolean requireVisit;
    private Boolean requireQuiz;
    private Boolean visitCompleted;
    private Boolean quizCompleted;
    private Boolean requirementsSatisfied;
    private Boolean completed;
    private Boolean overdue;
    private StudentDTO student;
}

File: Frontend/app/src/main/java/com/example/localhistory/model/response/HomeworkAssignmentDTO.kt
package com.example.localhistory.model.response

import kotlinx.datetime.LocalDate
import kotlinx.datetime.LocalDateTime

data class HomeworkAssignmentDTO(
    val id: Long,
    val homeworkId: Long,
    val title: String,
    val description: String,
    val dueDate: LocalDate,
    val assignedAt: LocalDateTime?,
    val completedAt: LocalDateTime?,
    val landmarkId: Long,
    val landmarkTitle: String?,
    val requireVisit: Boolean,
    val requireQuiz: Boolean,
    val visitCompleted: Boolean,
    val quizCompleted: Boolean,
    val requirementsSatisfied: Boolean,
    val completed: Boolean,
    val overdue: Boolean,
    val student: User.Student?
)
